pkg/bpf/collection.go

```diff
@@ -123,12 +124,40 @@ func LoadCollection(spec *ebpf.CollectionSpec, opts ebpf.CollectionOptions) (*eb
 		return nil, fmt.Errorf("inlining global data: %w", err)
 	}
 
-	coll, err := ebpf.NewCollectionWithOptions(spec, opts)
-	if err != nil {
-		return nil, err
+	// Set initial size of verifier log buffer.
+	if opts.Programs.LogSize == 0 {
+		opts.Programs.LogSize = 4_194_304
 	}
 
-	return coll, nil
+	attempt := 1
```

Comment:
I would probably do smth like
```
sizes := slice{4,16,64,...}
for size in sizes {
  try size
}
```
so that we explicitly show our intents without `attempts`, but this is just my CHF0.05

Reply:
Sure, but the goal was to avoid hardcoding sizes as much as possible and allow the caller to set the starting point of LogSize. (e.g. from tests)

Thanks for the review!

Reply:
Ah, sure, my bad